### ISSUE TITLE
Song with # Character in the file name does not play

### ISSUE THREAD

**Issue by murugadossp** [florent37/Flutter-AssetsAudioPlayer#251]
**Flutter Version**

My version : 
Flutter 1.18.0-11.1.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 2738a1148b (6 weeks ago) • 2020-05-13 15:24:36 -0700
Engine • revision ef9215ceb2
Tools • Dart 2.9.0 (build 2.9.0-8.2.beta)

**Lib Version**

My version : 
assets_audio_player 2.0.6+4

**Platform (Android / iOS / web) + version**

Platform :  Android

Trying to play

/storage/emulated/0/Download/y2mate.com - Night Melodies 27122019 _ #Ilaiyaraaja #ilaiyaraajahits #ilaiyaraajatamilhits #live_b3LX55yQ5RY.mp3 exists. 
The above file exists on the device checked with fileexists operation
However when given it to "Open" Method End up below issue.

W/os.music_playe(10489): Accessing hidden method Landroid/media/AudioTrack;->getLatency()I (greylist, reflection, allowed)
I/ExoPlayerImpl(10489): Init 5e9a197 [ExoPlayerLib/2.11.4] [OnePlus7Pro, GM1911, OnePlus, 29]
E/ExoPlayerImplInternal(10489): Source error
E/ExoPlayerImplInternal(10489):   com.google.android.exoplayer2.upstream.FileDataSource$FileDataSourceException: com.google.android.exoplayer2.upstream.FileDataSource$FileDataSourceException: uri has query and/or fragment, which are not supported. Did you call Uri.parse() on a string containing '?' or '#'? Use Uri.fromFile(new File(path)) to avoid this. path=/storage/emulated/0/Download/y2mate.com - Night Melodies 27122019 _ ,query=null,fragment=Ilaiyaraaja #ilaiyaraajahits #ilaiyaraajatamilhits #live_b3LX55yQ5RY.mp3

Caused by: com.google.android.exoplayer2.upstream.FileDataSource$FileDataSourceException: uri has query and/or fragment, which are not supported. Did you call Uri.parse() on a string containing '?' or '#'? Use Uri.fromFile(new File(path)) to avoid this. path=/storage/emulated/0/Download/y2mate.com - Night Melodies 27122019 _ ,query=null,fragment=Ilaiyaraaja #ilaiyaraajahits #ilaiyaraajatamilhits #live_b3LX55yQ5RY.mp3

which is resulting in 

E/ExoPlayerImplInternal(10489):   Caused by: java.io.FileNotFoundException: /storage/emulated/0/Download/y2mate.com - Night Melodies 27122019 _ : open failed: ENOENT (No such file or directory)

Please help to use the API Suggested : Uri.fromFile(new File(path))

After this _assetsAudioPlayer object goes into weird state, we need to recreate the Object

**Small code to reproduce**

```dart
import 'package:assets_audio_player/assets_audio_player.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State {
  
  final AssetsAudioPlayer _assetsAudioPlayer = AssetsAudioPlayer();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: RaisedButton(
            child: Text("open"),
            onPressed: () {
              //open code here
            }
          ),
        ),
      ),
    );
  }
}
```

**Comment by florent37** [comment]
thanks for the issue, I'll try to use `Uri.fromFile(new File(path))` instead of the uri

**Comment by florent37** [comment]
published in `2.0.7+1` can you try ? 

I tried to add a # in the file name in my `main_local_file.dart` and it works :)
